oneview/resource_server_hardware.go

```diff
@@ -200,14 +202,32 @@ func resourceServerHardwareCreate(d *schema.ResourceData, meta interface{}) erro
 }
 
 func resourceServerHardwareRead(d *schema.ResourceData, meta interface{}) error {
+	var (
+		servHard ov.ServerHardware
+		err      error
+	)
 	config := meta.(*Config)
 
-	servHard, err := config.ovClient.GetServerHardwareByName(d.Get("hostname").(string))
+	// fetching server hardware hostname incase it's added
+	if val, ok := d.GetOk("hostname"); ok {
+		servHard, err = config.ovClient.GetServerHardwareByName(val.(string))
+	} else {
+		// for refreshing imported server hardware we would need it's name
+		if val, ok := d.GetOk("name"); ok {
```

Comment:
When name attribute is Computed, how do we update TF configuration file with this attribute? If we are importing using the name, then we can directly refresh the resource without name. Whats the use of this block?

Reply:
name is computed so once a user import any SH name will get update and we are setting ID as UUID. 
so for 2nd time plan, we are using name from state to do a GET operation.